satpy/modifiers/parallax.py

```diff
@@ -0,0 +1,408 @@
+        nothing.  If only part of ``base_area`` is covered by ``cth_dataset``,
+        raise a `IncompleteHeightWarning`.  If none of ``base_area`` is covered
+        by ``cth_dataset``, raise a `MissingHeightError`.
+        """
+        warnings.warn(
+            "Overlap checking not impelemented. Waiting for "
+            "fix for https://github.com/pytroll/pyresample/issues/329")
+
+    def _invert_lonlat(self, pixel_lon, pixel_lat, source_area):
+        """Invert the lon/lat coordinate transformation.
+
+        When a satellite observes a cloud, a reprojection onto the cloud has
+        already happened.
+        """
+        (source_lon, source_lat) = source_area.get_lonlats(chunks=1024)
+        lon_diff = source_lon - pixel_lon
+        lat_diff = source_lat - pixel_lat
+        # We use the bucket resampler here, because parallax correction
+        # inevitably means there will be 2 source pixels ending up in the same
+        # destination pixel.  We want to choose the biggest shift (max abs in
+        # lat_diff and lon_diff), because the biggest shift corresponds to the
+        # highest clouds, and if we move a 10 km cloud over a 2 km one, we
+        # should retain the 10 km.
+        #
+        # some things to keep in mind:
+        # - even with a constant cloud height, 3 source pixels may end up in
+        #   the same destination pixel, because pixels get larger in the
+        #   direction of the satellite.  This means clouds may shrink as they
+        #   approach the satellite.
+        # - the x-shift is a function of y and the y-shift is a function of x,
+        #   so a cloud that was rectangular at the start may no longer be
+        #   rectangular at the end
+        br = BucketResampler(self.base_area,
+                             da.array(source_lon), da.array(source_lat))
+        inv_lat_diff = br.get_abs_max(lat_diff)
+        inv_lon_diff = br.get_abs_max(lon_diff)
+
+        (base_lon, base_lat) = self.base_area.get_lonlats(chunks=1024)
+        inv_lon = base_lon - inv_lon_diff
+        inv_lat = base_lat - inv_lat_diff
+        if self.debug_mode:
+            self.diagnostics["source_lon"] = source_lon
+            self.diagnostics["source_lat"] = source_lat
+            self.diagnostics["inv_lon"] = inv_lon
+            self.diagnostics["inv_lat"] = inv_lat
+            self.diagnostics["base_lon"] = base_lon
+            self.diagnostics["base_lat"] = base_lat
+            self.diagnostics["inv_lon_diff"] = inv_lon_diff
+            self.diagnostics["inv_lat_diff"] = inv_lat_diff
+            self.diagnostics["pixel_lon"] = pixel_lon
+            self.diagnostics["pixel_lat"] = pixel_lat
+            self.diagnostics["lon_diff"] = lon_diff
+            self.diagnostics["lat_diff"] = lat_diff
+            self.diagnostics["source_area"] = source_area
+            self.diagnostics["count"] = xr.DataArray(
+                br.get_count(), dims=("y", "x"), attrs={"area": self.base_area})
```

Comment:
It could, but it would need to pass a lot of locals, because it's essentially storing all the local variables, so I'm not sure if it would be an improvement in this case.

Reply:
Good point. Would it be possible to make them instance attributes? Or would that waste memory if `debug=False`?

Reply:
I prefer a dictionary, because it gathers all the debug information in a single dictionary that can be passed around and possibly modified by other methods.